### PR TITLE
Added the ability to disable automations

### PR DESCRIPTION
**Pull request by ConorWebb96**
## Description
This pr adds the ability to disable entire automations. As sometimes a user wants to test only or hasn't completely finished what the automation is supposed to do. Currently to prevent certain automations from running you have to delete it. Or avoid publishing the automation until you are finished creating it.

## Addresses
- [Disabling automations](https://github.com/Budibase/budibase/issues/13532)

## Screenshots
Screenshot 2024-05-13 at 09 14 14

## Launchcontrol
Add the ability to disable or enable automations.

### CODE REVIEW

File: packages/builder/src/components/design/settings/controls/ButtonActionEditor/actions/TriggerAutomation.svelte [hunk 1]
```diff
@@ -24,7 +24,11 @@
     parameters
   }
   $: automations = $automationStore.automations
-    .filter(a => a.definition.trigger?.stepId === TriggerStepID.APP)
+    .filter(
+      a =>
+        a.definition.trigger?.stepId === TriggerStepID.APP &&
+        a.disabled !== true
```
Comment (mike12345567) [inline packages/builder/src/components/design/settings/controls/ButtonActionEditor/actions/TriggerAutomation.svelte:30]: `!disabled` is usually preferred for flags, incase it were to be set to `1` for example

File: packages/server/src/automations/triggers.ts [hunk 2]
```diff
@@ -36,10 +36,14 @@ async function queueRelevantRowAutomations(
...
 
-    // filter down to the correct event type
+    // filter down to the correct event type and enabled automations
     automations = automations.filter(automation => {
       const trigger = automation.definition.trigger
-      return trigger && trigger.event === eventType
+      return (
+        trigger &&
+        trigger.event === eventType &&
+        (automation.disabled === undefined || automation.disabled === false)
```
Comment (mike12345567) [inline packages/server/src/automations/triggers.ts:45]: Again, probably no need for such a specific check, can `!disabled` be used here?

**Comment by ConorWebb96** [comment]
Minor UX amends -- likely one more to follow.

## Activated automation screenshot
Screenshot 2024-05-13 at 14 56 38

## Paused Automation screenshot
Screenshot 2024-05-13 at 14 56 53

**Review by shogunpurple — APPROVED** [review]
LGTM!

**Review by mike12345567 — APPROVED** [review]

**Pull request merged**